sort/bucketsort.go

```diff
@@ -0,0 +1,46 @@
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+// sorting an array using buckets
+// mainly useful when input is uniformly distributed over a range
```

Comment:
```suggestion
// BucketSort sorts a slice. It is mainly useful
// when input is uniformly distributed over a range.
```

Go style comments require the first word in a comment to be the same as the "thing" you're documenting.